src/components/piece/useCoordinate.js

```diff
@@ -0,0 +1,19 @@
+import { useState } from 'react';
+
+export const useCoordinate= () => {
+  const [coordinate, setCoordinate] = useState({cx: 75/2 , cy: 75/2, counter: 1});
```

Comment:
we need to review this, but i am suspicious about the "stateful" ness about this. it seems like every piece will have a coordinate. but this hook is just one coordinate. so every component that uses it will have the same coordinate.